modules/fundamental/src/advisory/endpoints/mod.rs

```diff
@@ -135,12 +135,26 @@ pub async fn upload(
 )]
 #[get("/v1/advisory/{key}/download")]
 pub async fn download(
-    service: web::Data<IngestorService>,
+    ingestor: web::Data<IngestorService>,
+    advisory: web::Data<AdvisoryService>,
     key: web::Path<String>,
 ) -> Result<impl Responder, Error> {
-    let hash_key = HashOrUuidKey::from_str(&key).map_err(Error::HashKey)?;
+    let hash_key = Id::from_str(&key).map_err(Error::HashKey)?;
+
+    let Some(adv) = advisory.fetch_advisory(hash_key, ()).await? else {
+        return Ok(HttpResponse::NotFound().finish());
+    };
+
+    let Some(hash) = adv.head.hashes.into_iter().find_map(|hash| match hash {
+        Id::Sha256(hash) => Some(hash),
+        _ => None,
+    }) else {
+        return Ok(HttpResponse::NotFound().finish());
+    };
+
+    let hash_key = Id::Sha256(hash);
```

Comment:
minor style nit. might warrant adding a `find_hash` method to `AdvisoryHead`

```suggestion
    let id = Id::from_str(&key).map_err(Error::HashKey)?;
    let Some(hash_key) = advisory.fetch_advisory(id, ()).await?.and_then(|adv| {
        adv.head
            .hashes
            .into_iter()
            .find(|h| matches!(h, Id::Sha256(_)))
    }) else {
        return Ok(HttpResponse::NotFound().finish());
    };
```